aggregator/aggregator.go

```diff
@@ -192,14 +221,30 @@ func (agg *aggregator) Open() error {
 	defer agg.Unlock()
 
 	if agg.state != aggregatorNotOpen {
-		return errAggregatorIsOpenOrClosed
+		return errAggregatorAlreadyOpenOrClosed
```

Comment:
`NotOpen` is the uninitialized state, which means the aggregator has never been opened before. Right now the semantics is that if the aggregator has been closed (which usually happens during a process shutdown), it cannot be re-opened. That semantics is hard to achieve with only two states.